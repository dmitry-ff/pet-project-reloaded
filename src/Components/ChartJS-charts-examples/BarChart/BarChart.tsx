import React from 'react'
import * as Styled from '../ChartWrapper.styled'
import { Bar } from "react-chartjs-2";
import { TUserData } from "../types/TUserData";
import { EmotionJSX } from "@emotion/react/types/jsx-namespace";

const BarChart: React.FC<TUserData> = ({ data }): EmotionJSX.Element => {
  return (
    <Styled.ChartWrapper>
      <Bar data={data} />
    </Styled.ChartWrapper>
  )
}

export default BarChart