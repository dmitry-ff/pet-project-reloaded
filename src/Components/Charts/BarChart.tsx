import { Bar } from "react-chartjs-2";
import ChartWrapper from "../../Style/ChartWrapper";
import { TUserData } from '../../Data/Types'

import React from 'react'

function BarChart({ data }: TUserData) {
  return (
    <ChartWrapper>
      <Bar data={data} />
    </ChartWrapper>
  )
}

export default BarChart