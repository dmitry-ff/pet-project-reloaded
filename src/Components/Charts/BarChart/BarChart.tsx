import { Bar } from "react-chartjs-2";
import ChartWrapper from '../ChartWrapper'
import { TUserData } from "../../../Data/Types/TUserData";

import React from 'react'
import { EmotionJSX } from "@emotion/react/types/jsx-namespace";

const BarChart: React.FC<TUserData> = ({ data }): EmotionJSX.Element => {
  return (
    <ChartWrapper>
      <Bar data={data} />
    </ChartWrapper>
  )
}

export default BarChart