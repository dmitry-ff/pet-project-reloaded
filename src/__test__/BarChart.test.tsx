import { shallow } from "enzyme";
import _ from "lodash";
import BarChart from "../Components/Charts/BarChart";
import renderer from 'react-test-renderer';

// import { userData } from "../Data/Data";
const userData = {

  labels: ['Red', 'Blue', 'Yellow',],
  datasets: [
    {
      label: 'Dataset 1',
      data: [1, 2, 3, 45, 6, 7],
      borderColor: 'rgb(255, 99, 132)',
      backgroundColor: 'rgba(255, 99, 132, 0.5)',
      tension: 0.4
    },
    {
      label: 'Dataset 2',
      data: _.reverse([1, 2, 3, 45, 6, 7]),
      borderColor: 'rgb(53, 162, 235)',
      backgroundColor: 'rgba(53, 162, 235, 0.5)',
      tension: 0.1
    },

  ]
}
describe('BarChart component', () => {
  it('shoud render BarChart', () => {
    const component = shallow(<BarChart data={userData} />)
    const wrapper = component.find('Styled(div)');
    expect(wrapper.length).toBe(1);
  });
  it('Barchart snapshot', () => {
    const tree = renderer
      .create(<BarChart data={userData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
export { }