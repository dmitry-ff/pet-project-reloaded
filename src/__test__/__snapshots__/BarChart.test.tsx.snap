// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart component Barchart snapshot 1`] = `
<div
  className="css-1469kb1"
>
  <canvas
    height={150}
    role="img"
    width={300}
  />
</div>
`;
